### ISSUE TITLE
[Android] Focus is removed from TextInput when Talkback is enabled

### ISSUE THREAD

**Issue by oargaruna** [facebook/react-native#12205]
### Description

Our project is a hybrid of React Native modules and native UI. We are adding accessibility to our App and are hitting a weird issue with Talkback in TextInput:

If there is any clickable element before a TextInput in the view hierarchy, double tapping to add focus to the TextInput using Talkback sends the focus to the clickable element before it.

### Reproduction

I have been able to reproduce the bug in a sample app: https://github.com/oargaruna/TextInputTalkback

1. Deploy the App on an Android device
2. Enable Talkback (from Settings -> Accessibility) and open the Sample App
3. Tap once on the TextInput to give it Talkback focus
4. Tap twice to try to get the keyboard to come up

We expect the keyboard to show up with focus on the TextInput, but the focus jumps to the TouchableOpacity element defined before it.

If the TouchableOpacity element is removed, this behaviour is not reproducible.

### Additional Information

* React Native version: 0.41.1
* Platform: Android

**Comment by JasonWang-MS** [comment]
Same issue for me. Any solution?

**Comment by hramos** [comment]
Hi there! This issue is being closed because it has been inactive for a while. Maybe the issue has been fixed in a recent release, or perhaps it is not affecting a lot of people. Either way, we're automatically closing issues after a period of inactivity. Please do not take it personally!

If you think this issue should definitely remain open, please let us know. The following information is helpful when it comes to determining if the issue should be re-opened:

- Does the issue still reproduce on the latest release candidate? Post a comment with the version you tested.
- If so, is there any information missing from the bug report? Post a comment with all the information required by the [issue template](https://github.com/facebook/react-native/blob/master/.github/ISSUE_TEMPLATE.md).
- Is there a pull request that addresses this issue? Post a comment with the PR number so we can follow up.

If you would like to work on a patch to fix the issue, *contributions are very welcome*! Read through the [contribution guide](http://facebook.github.io/react-native/docs/contributing.html), and feel free to hop into [#react-native](https://discordapp.com/invite/0ZcbPKXt5bZjGY5n) if you need help planning your contribution.

**Comment by projetolead** [comment]
This is still happening to me on RN 0.49.3. Any workarounds?

**Comment by lelandrichardson** [comment]
Reopening this as it appears to be affecting us as well.

**Comment by backwardok** [comment]
Looks like this might be a dupe of https://github.com/facebook/react-native/issues/17042 ?

**Comment by haitaoli** [comment]
The issue is caused by this line of code: https://github.com/facebook/react-native/blob/c9ff0bc212b680232f7379fba7b9332927075c3c/ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java#L202. It's been there forever and I haven't figured out why it's added.

**Comment by haitaoli** [comment]
Found some comments in code:
```  // This component is controlled, so we want it to get focused only when JS ask it to do so.
  // Whenever android requests focus (which it does for random reasons), it will be ignored.```
I wonder if this is still an issue in recent Android versions.

**Comment by oargaruna** [comment]
The issue is not reproducible in 0.51.

**Comment by haitaoli** [comment]
@oargaruna this only happens on Android 6.

**Comment by RamyaSenk** [comment]
It is not working on Android 8.0 on latest version of react native as well.

**Comment by NitroGhost** [comment]
still an issue

**Comment by jhta** [comment]
It's still happen in Android 8.0 - RN 0.55.4

**Comment by haitaoli** [comment]
I don't think this will ever be fixed in official releases. Fork RN and apply this PR: https://github.com/airbnb/react-native/pull/47.

**Comment by FLGMwt** [comment]
This is still affecting Android 8.0 - RN 0.57.1

Do we have any context for the hack? We'd prefer not to maintain a fork, especially since that was explicitly referenced in the AirBnbB posts : P

**Comment by BenWildeman** [comment]
still an issue, bot

**Comment by JJBarata** [comment]
Hello, I'm currently using Android 8.1.0 with RN 0.51.0 and RN 0.58.3 (two apps) and both have the same problem when talkback is enabled, the keyboard does not appear when autofocus is in the textinput.
Does anyone still have the same problem yet?
Another thing I noticed is that this is happening intermittently, that is, sometimes it works, sometimes it does not work.
Thanks.

**Comment by raphaelssampaio** [comment]
I have the same problem here.

**Comment by dishantwalia** [comment]
Double tap and hold to long press on TextInput keyboard will come up, it's default behaviour. Works for me

**Comment by BenWildeman** [comment]
@dishantwalia that is not default behaviour. everything should be accessible by double tapping.

>To explore by touch, slowly drag one finger around the screen. TalkBack announces the icons, buttons and other items as you drag your finger over them. When the focus reaches an item that you'd like to select, double tap anywhere on the screen to select the focused item.
https://support.google.com/accessibility/android/answer/6006598?hl=en-GB

you should make sure that whatever you're testing this against is actually an input field and not a wrapper. for instance, is you use the floating label input field type provided by Native Base, you can access the input field by double tapping on the wrapper

**Comment by fhugoduarte** [comment]
https://github.com/facebook/react-native/issues/24519

**Comment by BenWildeman** [comment]
still an issue

**Comment by BenWildeman** [comment]
still not fixed
